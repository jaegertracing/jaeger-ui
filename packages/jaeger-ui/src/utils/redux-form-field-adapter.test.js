// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { Input, Popover } from 'antd';
import { shallow } from 'enzyme';
import React from 'react';

import reduxFormFieldAdapter from './redux-form-field-adapter';

describe('reduxFormFieldAdapter', () => {
  const error = {
    content: 'error content',
    title: 'error title',
  };
  let input;
  let meta;

  beforeEach(() => {
    input = {
      onChange: function onChange() {},
      onBlur: function onBlur() {},
      value: 'inputValue',
    };
    meta = {
      error: null,
      active: false,
    };
  });

  describe('not validated input', () => {
    const AdaptedInput = reduxFormFieldAdapter({ AntInputComponent: Input });
    it('should render as expected', () => {
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('validate input', () => {
    const AdaptedInput = reduxFormFieldAdapter({ AntInputComponent: Input, isValidatedInput: true });
    it('should render as expected when there is not an error', () => {
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render Popover as invisible if there is an error but the field is active', () => {
      meta.error = error;
      meta.active = true;
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);
      expect(wrapper.find(Popover).prop('visible')).toBeFalsy();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render Popover as visible if there is an error and the field is not active', () => {
      meta.error = error;
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);
      expect(wrapper.find(Popover).prop('visible')).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('onChangeAdapter', () => {
    it('should use input.onChange when adapter is not provided', () => {
      const AdaptedInput = reduxFormFieldAdapter({ AntInputComponent: Input });
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);
      expect(wrapper.find(Input).prop('onChange')).toBe(input.onChange);
    });

    it('should call input.onChange with result of adapter when adapter is provided', () => {
      const onChangeMockInput = 'onChangeMockInput';
      const onChangeAdapterMockReturnValue = 'onChangeAdapterMockReturnValue';
      const onChangeAdapter = jest.fn().mockReturnValue(onChangeAdapterMockReturnValue);
      const onChangeSpy = jest.fn();
      input.onChange = onChangeSpy;
      const AdaptedInput = reduxFormFieldAdapter({ AntInputComponent: Input, onChangeAdapter });
      const wrapper = shallow(<AdaptedInput input={input} meta={meta} />);

      const renderedOnChangeFn = wrapper.find(Input).prop('onChange');
      expect(renderedOnChangeFn).not.toBe(input.onChange);
      renderedOnChangeFn(onChangeMockInput);
      expect(onChangeAdapter).toHaveBeenCalledWith(onChangeMockInput);
      expect(onChangeSpy).toHaveBeenCalledWith(onChangeAdapterMockReturnValue);
    });
  });
});
