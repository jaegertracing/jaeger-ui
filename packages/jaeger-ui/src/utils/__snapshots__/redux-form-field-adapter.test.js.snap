// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduxFormFieldAdapter not validated input should render as expected 1`] = `
<Input
  className=""
  disabled={false}
  meta={
    Object {
      "active": false,
      "error": null,
    }
  }
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  prefixCls="ant-input"
  type="text"
  value="inputValue"
/>
`;

exports[`reduxFormFieldAdapter validate input should render Popover as invisible if there is an error but the field is active 1`] = `
<Popover
  content="error content"
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayStyle={Object {}}
  placement="bottomLeft"
  prefixCls="ant-popover"
  title="error title"
  transitionName="zoom-big"
  trigger="hover"
  visible={false}
>
  <Input
    className="AdaptedReduxFormField--isValidatedInput"
    disabled={false}
    meta={
      Object {
        "active": true,
        "error": Object {
          "content": "error content",
          "title": "error title",
        },
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    prefixCls="ant-input"
    type="text"
    value="inputValue"
  />
</Popover>
`;

exports[`reduxFormFieldAdapter validate input should render Popover as visible if there is an error and the field is not active 1`] = `
<Popover
  content="error content"
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayStyle={Object {}}
  placement="bottomLeft"
  prefixCls="ant-popover"
  title="error title"
  transitionName="zoom-big"
  trigger="hover"
  visible={true}
>
  <Input
    className="is-invalid AdaptedReduxFormField--isValidatedInput"
    disabled={false}
    meta={
      Object {
        "active": false,
        "error": Object {
          "content": "error content",
          "title": "error title",
        },
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    prefixCls="ant-input"
    type="text"
    value="inputValue"
  />
</Popover>
`;

exports[`reduxFormFieldAdapter validate input should render as expected when there is not an error 1`] = `
<Popover
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayStyle={Object {}}
  placement="bottomLeft"
  prefixCls="ant-popover"
  transitionName="zoom-big"
  trigger="hover"
  visible={false}
>
  <Input
    className="AdaptedReduxFormField--isValidatedInput"
    disabled={false}
    meta={
      Object {
        "active": false,
        "error": null,
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    prefixCls="ant-input"
    type="text"
    value="inputValue"
  />
</Popover>
`;
